File: Python/Remove Nth Node From End of List .py
class Solution:
    # @return a ListNode
    def removeNthFromEnd(self, head, n):
        if not head:
            return None
        fast = head
        slow = head
        for i in range(n):
            fast = fast.next
        if (fast == None):
            return slow.next;

        while fast.next != None:
            fast = fast.next
            slow = slow.next
        slow.next = slow.next.next
        return head

File: Java/src/Remove Nth Node From End of List .java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if (head == null) return null;
        ListNode fast = head;
        ListNode slow = head;
        
        for (int i=0; i<n; i++){
            fast = fast.next;
        }
        if (fast == null) return slow.next;
        
        while (fast.next != null){
            fast = fast.next;
            slow = slow.next;
        }
        slow.next = slow.next.next;
        return head;
        
    }
}